onnx/helper.py

```diff
@@ -854,34 +857,57 @@ def make_attribute(  # pylint: disable=too-many-statements
     # Iterable cases
     elif isinstance(value, collections.abc.Iterable):
         value = list(value)
-        types = {type(v) for v in value}
-        if all(issubclass(t, numbers.Integral) for t in types):
+        if len(value) == 0 and attr_type is None:
+            raise ValueError("Could not infer attribute type from empty iterator")
```

Comment:
It would be nice to add the parameter name in the error message. It would be really helpful when writing a converter.